src/web/routes.rs

```diff
@@ -204,15 +220,26 @@ impl Routes {
     }
 }
 
-/// Simple Iron route that redirects to the same path without the trailing slash, for example
-/// /foo/bar/baz/ to /foo/bar/baz.
-fn remove_slashes(req: &mut iron::Request) -> iron::IronResult<iron::Response> {
-    let mut url: iron::url::Url = req.url.clone().into();
-    url.set_path(&url.path().trim_end_matches('/').to_string());
-    Ok(iron::Response::with((
-        iron::status::Found,
-        iron::modifiers::Redirect(iron::Url::parse(&url.to_string()).unwrap()),
-    )))
+#[derive(Copy, Clone)]
+struct SimpleRedirect {
+    url_mangler: fn(&mut iron::url::Url),
```

Comment:
Not super important, but is there a reason you made this a struct instead of a trait with a default implementation for `Handler`? That would allow static instead of dynamic dispatch.

Reply:
Ignore this - if it's a trait than it's really annoying to call `SimpleRedirect::new`

Reply:
Hmm? I'm not sure I follow what you're proposing.